tests/pstl/move.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of AdaptiveCpp, an implementation of SYCL and C++ standard
+ * parallelism for CPUs and GPUs.
+ *
+ * Copyright The AdaptiveCpp Contributors
+ *
+ * AdaptiveCpp is released under the BSD 2-Clause "Simplified" License.
+ * See file LICENSE in the project root for full license details.
+ */
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+#include <execution>
+#include <utility>
+#include <vector>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/mpl/list.hpp>
+
+#include "pstl_test_suite.hpp"
+
+BOOST_FIXTURE_TEST_SUITE(pstl_move, enable_unified_shared_memory)
+
+template<class T, class Policy>
+void test_move(Policy&& pol, std::size_t problem_size) {
+    std::vector<T> data(problem_size);
+    for (int i = 0; i < problem_size; ++i) {
+        data[i] = T{i};
+    }
+
+    std::vector<T> dest_device(problem_size);
+    std::vector<T> dest_host(problem_size);
+
+    auto ret = std::move(pol, data.begin(), data.end(),
+                         dest_device.begin());
+    std::move(data.begin(), data.end(), dest_host.begin());
+
+    BOOST_CHECK(ret == dest_device.begin() + problem_size);
+    BOOST_CHECK(dest_device == dest_host);
+    BOOST_CHECK(data == dest_device);
+    BOOST_CHECK(data == dest_host);
```

Comment:
Is this a valid check? It is not guaranteed that the elements of the test vector remain intact after a move operation. I could not come up with a rigorous test

Reply:
That's right, this might be problematic, depending on what type `T` is. You could just copy the original `data` though before you run the test:

```
std::vector<T> data(problem_size)
// init data

auto device_input = data;
auto host_input = data;

auto ret = std::move(pol, device_input.begin(), device_input.end(),
                         dest_device.begin());
std::move(host_input.begin(), host_output.end(), dest_host.begin());
```